update testing module (#204)
school-brainhack/school-brainhack.github.io: content/en/modules/testing/index.md

 * the [Using git and github](/modules/git_github) module.
 * the [A brief introduction to the bash shell](/modules/introduction_to_terminal) module.

> It is highly recommended to also have done the [python script](/modules/python_scripts) modue, but that's not strictly necessary.

Reviewer: ```suggestion
It is highly recommended to also have done the [python script](/modules/python_scripts) module, but that's not strictly necessary.
```

Contact François Paugam if you have questions on this module, or if you want to check that you completed successfully all the exercises.

## Resources
The [slides](https://drive.google.com/file/d/1M3nr4D0-cPCjHjL23vk-BXth2TqHshtj/view?usp=sharing) are adapted from the original QLSC 612 course in 2020 by [Greg Kiar](https://twitter.com/g_kiar)

The video of the presentation is available below:
<iframe width="560" height="315" src="https://www.youtube.com/embed/TIPIap8rZyE" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>

## Exercise

>  * Fork [this repo](https://github.com/FrancoisPgm/testing_CI_module) for the hands on part.

Reviewer: What do you think if we have a fork on the brainhack school organisation, so it's easier for future maintenance? Like we don't need to bug you every year

Author: yes sure

>  * Watch the video, and follow along the hands on implementing the unit tests and the github action to run them.

Reviewer: ```suggestion
 * Watch the video, and follow along the hands on material. You will implement the unit tests and the github action to execute the unit test.
```

 * Follow up with François Paugam to validate you completed the exercise correctly.
 * :tada: :tada: :tada: you completed this training module! :tada: :tada: :tada:

>  **Bonus and addendum** : During the first part of the video I said we would implement unit tests, an integration test and an installation test. I forgot about the integration test during the hands on so I leave it t you as a bonus exercise, which corresponds to writing a test for the main function, since it integrates the other three functions. For the installation test, this one comes for free with the github action, as the action runs in a virtual machine and has to re-install the dependancies each time. 

Reviewer: ```suggestion
 **Bonus and addendum** : During the first part of the video I said we would implement unit tests, an integration test and an installation test. I forgot about the integration test during the hands on so I leave it to you as a bonus exercise, which corresponds to writing a test for the main function, since it integrates the other three functions. For the installation test, this one comes for free with the github action, as the action runs in a virtual machine and has to re-install the dependencies each time. 
```
